Libraries/LibWeb/CSS/StyleValues/ShorthandStyleValue.cpp

```diff
@@ -107,14 +107,39 @@ String ShorthandStyleValue::to_string(SerializationMode mode) const
         auto origin = longhand(PropertyID::BackgroundOrigin);
         auto clip = longhand(PropertyID::BackgroundClip);
 
+        auto serialize_layer = [mode](Optional<String> color_value_string, String image_value_string, String position_x_value_string, String position_y_value_string, String size_value_string, String repeat_value_string, String attachment_value_string, String origin_value_string, String clip_value_string) {
+            StringBuilder builder;
+
+            Vector<PropertyID> property_ids = { PropertyID::BackgroundColor, PropertyID::BackgroundImage, PropertyID::BackgroundPositionX, PropertyID::BackgroundPositionY, PropertyID::BackgroundSize, PropertyID::BackgroundRepeat, PropertyID::BackgroundAttachment, PropertyID::BackgroundOrigin, PropertyID::BackgroundClip };
+            Vector<Optional<String>> property_value_strings = { move(color_value_string), move(image_value_string), move(position_x_value_string), move(position_y_value_string), move(size_value_string), move(repeat_value_string), move(attachment_value_string), move(origin_value_string), move(clip_value_string) };
+
+            for (size_t i = 0; i < property_ids.size(); i++) {
+                if (!property_value_strings[i].has_value())
+                    continue;
+
+                auto intial_property_string_value = property_initial_value(property_ids[i])->to_string(mode);
+
+                if (property_value_strings[i].value() != intial_property_string_value) {
+                    if (!builder.is_empty())
+                        builder.append(" "sv);
+                    builder.append(property_value_strings[i].value());
+                }
+            }
+
+            if (builder.is_empty())
+                return "none"_string;
+
+            return MUST(builder.to_string());
```

Comment:
We know the builder only contains valid utf-8, so can use `builder.to_string_without_validation()` instead.